block/src/Pos/Block/Dump.hs

```diff
@@ -0,0 +1,273 @@
+{-# LANGUAGE RankNTypes #-}
+
+{- | Block dumps (used for snapshots).
+
+A block dump looks like this:
+
+@
+[block version, [block1, block2, ...]]
+@
+
+In CBOR terms, this is translated into this sequence of tokens:
+
+@
+listlen=2, block version, list_indef_len, [block1] [block2] ...
+@
+
+When we encode blocks, we strip proofs from them. When we decode blocks, we
+reconstruct those proofs.
+
+-}
+
+module Pos.Block.Dump
+       (
+       -- * Encoding
+         encodeBlockDump
+       , encodeBlockDump'
+
+       -- * Decoding
+       , decodeBlockDump
+       , decodeBlockDump'
+       , BlockDumpDecodeError (..)
+
+       -- * Internals
+       -- ** Encoding
+       , encodeDumpHeader
+       , encodeBlockStream
+       -- ** Decoding
+       , decodeDumpHeader
+       , decodeBlockStream
+       -- ** Stripped blocks
+       , GenericBlockNoProof (..)
+       , BlockNoProof
+       , GenesisBlockNoProof
+       , MainBlockNoProof
+       , stripProof
+       , addProof
+       ) where
+
+import           Universum
+
+import qualified Codec.CBOR.Decoding as Cbor hiding (DecodeAction (..))
+import qualified Codec.CBOR.Encoding as Cbor
+import qualified Codec.CBOR.Read as Cbor
+import qualified Codec.CBOR.Write as Cbor
+import           Conduit (Conduit, Consumer, Producer, runConduit, yield, (.|))
+import qualified Conduit
+import qualified Data.Conduit.Lzma as Lzma
+import qualified Data.Text.Buildable
+import           Formatting (bprint, int, stext, (%))
+import           Serokell.Util (listJson)
+
+import           Pos.Binary.Class (Bi (..), serialize')
+import qualified Pos.Binary.Class as Bi
+import qualified Pos.Binary.Conduit as Bi
+import           Pos.Core (Block, GenericBlock (..), GenericBlockHeader (..))
+import qualified Pos.Core as C
+import           Pos.Crypto (HasCryptoConfiguration)
+import           Pos.Util.Chrono (OldestFirst (..))
+import           Pos.Util.Util (eitherToFail)
+
+----------------------------------------------------------------------------
+-- Encoding
+----------------------------------------------------------------------------
+
+-- | Encode a block dump. The blocks should be coming oldest-first.
+encodeBlockDump
+    :: (Bi BlockNoProof, Conduit.MonadResource m)
+    => Conduit Block m ByteString
+encodeBlockDump = (encodeDumpHeader >> encodeBlockStream)
+               .| Lzma.compress Nothing
+
+-- | Like 'encodeBlockDump', but without conduits.
```

Comment:
Please follow your naming convention everywhere. Earlier you defined `func` and `funcC` for a non-conduit and conduit version respectively. Can you rename these functions to follow this convention?